scan-payment/src/main/java/com/getbouncer/scan/payment/analyzer/PaymentCardOcrAnalyzer.kt

```diff
@@ -2,31 +2,32 @@ package com.getbouncer.scan.payment.analyzer
 
 import com.getbouncer.scan.framework.Analyzer
 import com.getbouncer.scan.framework.AnalyzerFactory
+import com.getbouncer.scan.payment.ml.ExpiryDetect
 import com.getbouncer.scan.payment.ml.SSDOcr
 import com.getbouncer.scan.payment.ml.ssd.DetectionBox
 import kotlinx.coroutines.async
 import kotlinx.coroutines.supervisorScope
 
 class PaymentCardOcrAnalyzer private constructor(
     private val ssdOcr: SSDOcr?,
-    private val nameDetect: NameDetectAnalyzer?
+    private val nameAndExpiryAnalyzer: NameAndExpiryAnalyzer?
 ) : Analyzer<SSDOcr.Input, PaymentCardOcrState, PaymentCardOcrAnalyzer.Prediction> {
 
     data class Prediction(
         val pan: String?,
         val panDetectionBoxes: List<DetectionBox>?,
         val name: String?,
-        val expiry: String?,
+        val expiry: ExpiryDetect.Expiry?,
         val objDetectionBoxes: List<DetectionBox>?,
-        val isNameExtractionAvailable: Boolean
+        val isNameAndExpiryExtractionAvailable: Boolean
     )
 
     override val name: String = "payment_card_ocr_analyzer"
 
     override suspend fun analyze(data: SSDOcr.Input, state: PaymentCardOcrState) = supervisorScope {
-        val cardDetectFuture = if (state.runNameExtraction && nameDetect != null) {
+        val cardDetectFuture = if ((state.runNameExtraction || state.runExpiryExtraction) && nameAndExpiryAnalyzer != null) {
```

Comment:
nit: `Future` is a different kind of class that we don't currently use in this project. the `async` call actually returns a `Deferred`.

Reply:
I realize I made this mistake in the CardSideAnalyzer. I'll make a note to rename that myself.